src/world/footpath.c

```diff
@@ -1740,9 +1740,9 @@ void footpath_update_path_wide_flags(int x, int y)
 	footpath_clear_wide(x, y);
 	y -= 0x20;
 
-	if (!(x & 0xE0))
+	if (!(x & 0xFFE0))
 		return;
-	if (!(y & 0xE0))
+	if (!(y & 0xFFE0))
```

Comment:
Regarding the change to line 1745.  Interesting that this resolves some paths getting disconnected.  The intent in this condition was to add in extra non-wide tiles (overlaying a grid of non-wide tiles onto the path network) - I presume the intent was to encourage peeps to walk around more in large areas that are "paved". I certainly wouldn't expect this to cause such issues, but the evidence cannot be denied. So, sheer fluke that by changing this I fixed the issue with disconnected paths.  And you are quite right that those lines can be removed completely, so I'll do that.